docs/getting-started/provider-azure.md

```diff
@@ -0,0 +1,216 @@
+Follow the documentation from Microsoft to [Download and install the Azure command-line](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+Log in to the Azure command-line.
+
+```command
+az login
+```
+#### Create an Azure service principal
+Follow the Azure documentation to [find your Subscription ID](https://docs.microsoft.com/en-us/azure/azure-portal/get-subscription-tenant-id) from the Azure Portal.
+
+Using the Azure command-line and provide your Subscription ID create a service principal and authentication file.
+
+```shell {copy-lines="all"}
+az ad sp create-for-rbac \
+--sdk-auth \
+--role Owner \
+--scopes /subscriptions/<Subscription ID> 
+```
+
+Save your Azure JSON output as `azure-credentials.json`.
+
+{{< hint type="note" >}}
+The [Configuration](https://marketplace.upbound.io/providers/upbound/provider-azure/latest/docs/configuration) section of the Provider documentation describes other authentication methods.
+{{< /hint >}}
+
+#### Create a Kubernetes secret with the Azure credentials
+A Kubernetes generic secret has a name and contents. Use {{< hover label="kube-create-secret" line="1">}}kubectl create secret{{< /hover >}} to generate the secret object named {{< hover label="kube-create-secret" line="2">}}azure-secret{{< /hover >}} in the {{< hover label="kube-create-secret" line="3">}}crossplane-system{{</ hover >}} namespace.  
+
+<!-- vale gitlab.Substitutions = NO -->
+<!-- ignore .json file name -->
+Use the {{< hover label="kube-create-secret" line="4">}}--from-file={{</hover>}} argument to set the value to the contents of the  {{< hover label="kube-create-secret" line="4">}}azure-credentials.json{{< /hover >}} file.
+<!-- vale gitlab.Substitutions = YES -->
+```shell {label="kube-create-secret",copy-lines="all"}
+kubectl create secret \
+generic azure-secret \
+-n crossplane-system \
+--from-file=creds=./azure-credentials.json
+```
+
+View the secret with `kubectl describe secret`
+
+{{< hint type="note" >}}
+The size may be larger if there are extra blank spaces in your text file.
+{{< /hint >}}
+
+```shell
+kubectl describe secret azure-secret -n crossplane-system
+Name:         azure-secret
+Namespace:    crossplane-system
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+creds:  629 bytes
+```
+
+### Create a ProviderConfig
+A `ProviderConfig` customizes the settings of the Azure Provider.  
+
+Apply the {{< hover label="providerconfig" line="2">}}ProviderConfig{{</ hover >}} with the command:
+```yaml {label="providerconfig",copy-lines="all"}
+cat <<EOF | kubectl apply -f -
+apiVersion: azure.upbound.io/v1beta1
+metadata:
+  name: default
+kind: ProviderConfig
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: azure-secret
+      key: creds
+EOF
```

Comment:
This isn't valid YAML; it's a snippet of shell.

Reply:
Reasoning has been added to the [contributing guide](https://github.com/crossplane/crossplane/pull/3396/files#diff-93426d953f03b733adbbb574ffd2717d0892773a464c0d301d1b6c9095d67b15R191).

Resolving this conversation (unresolve if you're not satisfied with this)